tests/integration/integration-test.py

```diff
@@ -34,73 +38,141 @@ def setup(cls, config_path, openapi_path):
     def cleanup(cls):
         cls.session.close()
 
-    # Test case: GET /pets
-    def test_get_all_pets(self, endpoint='/pets'):
-        nullable_fields = ['owner']
-        utils.test_endpoint(self, endpoint, 'PetResource', 200,
-                            nullable_fields=nullable_fields)
-
-    # Test case: GET /pets with species filter
-    def test_get_pets_with_filter(self, endpoint='/pets'):
-        testing_species = ['dog', 'CAT', 'tUrTlE']
+    # Test GET /parks
+    def test_get_all_parks(self, endpoint='/parks'):
+        utils.test_endpoint(self, endpoint, 'ParkResource', 200)
+        valid_amenities = self.test_cases['valid_amenities']
+        invalid_amenities = self.test_cases['invalid_amenities']
 
-        for species in testing_species:
-            params = {'species': species}
-            response = utils.test_endpoint(self, endpoint, 'PetResource', 200,
+        # Test filter[amenities][all]
+        for amenities in valid_amenities:
+            params = {'filter[amenities][all]': amenities}
+            response = utils.test_endpoint(self, endpoint, park_res, 200,
+                                           query_params=params)
+            response_data = response.json()['data']
+            for resource in response_data:
+                actual_amenities = resource['attributes']['amenities']
+                for amenity in amenities.split(','):
+                    self.assertIn(amenity, actual_amenities)
+        for amenities in invalid_amenities:
+            params = {'filter[amenities][all]': amenities}
+            utils.test_endpoint(self, endpoint, 'ErrorObject', 400,
+                                query_params=params)
+
+        # Test filter[amenities][some]
+        for amenities in valid_amenities:
+            amen_list = amenities.split(',')
+            params = {'filter[amenities][some]': amenities}
+            response = utils.test_endpoint(self, endpoint, park_res, 200,
+                                           query_params=params)
+            response_data = response.json()['data']
+            for resource in response_data:
+                actual_amenities = resource['attributes']['amenities']
+                self.assertTrue(
+                    # asserts if at least one item in amen_list is also in
+                    # actual_amenities
+                    any(True for x in actual_amenities if x in amen_list)
+                )
+        for amenities in invalid_amenities:
+            params = {'filter[amenities][all]': amenities}
+            utils.test_endpoint(self, endpoint, err_obj, 400,
+                                query_params=params)
```

Comment:
Should this be `filter[amenities][some]`? Besides the filter operation( `all` and `some`), the logic here is exactly the same as the `filter[amenities][all]`. Do you think we can use a helper function to combine them?